src/main/java/org/opensearch/jobscheduler/rest/action/RestReleaseLockAction.java

```diff
@@ -0,0 +1,140 @@
+        );
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient nodeClient) throws IOException {
+        String lockId = restRequest.param(LOCK_ID);
+        if (lockId == null || lockId.isEmpty()) {
+            throw new IOException("lockId cannot be null or empty");
+        }
+
+        CompletableFuture<LockModel> findInProgressFuture = new CompletableFuture<>();
+        lockService.findLock(lockId, ActionListener.wrap(lock -> {
+            releaseLock = lock;
+            findInProgressFuture.complete(lock);
+        }, exception -> {
+            logger.error("Could not find lock model with lockId " + lockId, exception);
+            findInProgressFuture.completeExceptionally(exception);
+        }));
+
+        try {
+            findInProgressFuture.orTimeout(JobDetailsService.TIME_OUT_FOR_REQUEST, TimeUnit.SECONDS).join();
+        } catch (CompletionException e) {
+            if (e.getCause() instanceof TimeoutException) {
+                logger.info(" Request timed out with an exception ", e);
+            } else {
+                throw e;
+            }
+        } catch (Exception e) {
+            logger.info(" Could not find lock model with lockId due to exception ", e);
+        }
+
+        if (releaseLock != null) {
+            CompletableFuture<Boolean> releaseLockInProgressFuture = new CompletableFuture<>();
+            lockService.release(releaseLock, new ActionListener<>() {
+                @Override
+                public void onResponse(Boolean response) {
+                    releaseResponse = response;
+                    releaseLockInProgressFuture.complete(response);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    logger.error("Releasing lock failed with an exception", e);
+                    releaseLockInProgressFuture.completeExceptionally(e);
+                }
+            });
+
+            try {
+                releaseLockInProgressFuture.orTimeout(JobDetailsService.TIME_OUT_FOR_REQUEST, TimeUnit.SECONDS).join();
+            } catch (CompletionException e) {
+                if (e.getCause() instanceof TimeoutException) {
+                    logger.info(" Request timed out with an exception ", e);
+                } else {
+                    throw e;
+                }
+            } catch (Exception e) {
+                logger.info(" Could not release lock with " + releaseLock.getLockId() + " due to exception ", e);
+            }
+        }
+
+        return channel -> {
+            XContentBuilder builder = channel.newBuilder();
+            RestStatus restStatus = RestStatus.OK;
+            String restResponseString = releaseResponse ? "success" : "failed";
+            BytesRestResponse bytesRestResponse;
+            try {
+                builder.startObject();
+                builder.field("response", restResponseString);
```

Comment:
I would re-name the field to `release-lock` to be more explicit when the client reads it. `response` is generic and it bring ambiguity. 
```suggestion
                builder.field("release-lock", restResponseString);
```

Reply:
ACK